@here/olp-sdk-dataservice-read/lib/utils/getTile.ts

```diff
@@ -115,7 +119,9 @@ export async function getTile(
         catalogHrn,
         layerId,
         layerType
-    ).withQuadKey(quadKey);
+    )
+        .withQuadKey(quadKey)
+        .withDepth(depth);
```

Comment:
I think there is a misunderstanding here. If the user requests a tile, e.g. 1234556789 on level 14, you do not use this tile as quadKey base for the quad tree index call. You go to its parent on level 10 and request with that one with depth 4 so that you get all the neighbouring tiles of 1234556789 and the parents on level 13, the grandparents on level 12 and the grandgrandparents on level 11. If you request quad tree index with a level 14 tile with dept 4 you get 1 partition back.

So basically you need to do something like:

```suggestion
        .withQuadKey(computeParentKey(quadKey, depth))
        .withDepth(depth);
```

Reply:
Updated, thanks. Now we're caching the Quad Tree from parent of the requested tile with delta 4 and returning the blob of the requested Tile or the blob of the closest parent Tile. (the same datahandle)